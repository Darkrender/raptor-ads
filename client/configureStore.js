import { createStore, applyMiddleware, combineReducers } from 'redux';
import createLogger from 'redux-logger';
import rootReducer from './reducers';
import { routerReducer } from 'react-router-redux';

const reducers = combineReducers({
  app: rootReducer,
  routing: routerReducer
})

const configureStore = () => {
  const logger = createLogger();
  return createStore(reducers, applyMiddleware(logger));
}

export default configureStore;
