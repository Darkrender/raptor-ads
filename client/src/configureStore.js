import { createStore, applyMiddleware } from 'redux';
import createLogger from 'redux-logger';
import thunk from 'redux-thunk';
import { routerMiddleware } from 'react-router-redux';
import { browserHistory } from 'react-router';
import rootReducer from './reducers';

const middleware = routerMiddleware(browserHistory);

const configureStore = () => {
  if (process.env.NODE_ENV === 'production') {
    return createStore(rootReducer, applyMiddleware(thunk, middleware));
  }
  const logger = createLogger();
  return createStore(rootReducer, applyMiddleware(logger, thunk, middleware));
};

export default configureStore;
