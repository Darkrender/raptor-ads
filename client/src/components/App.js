import React from 'react';
import configureStore from '../../configureStore';
import { Provider } from 'react-redux';
import { Router, Route, browserHistory } from 'react-router';

const store = configureStore();

export const App = ({store}) =>
  <Provider store={store}>
    <Router>
      <Route path="/" component={HomePage} />
    </Router>
  </Provider>



