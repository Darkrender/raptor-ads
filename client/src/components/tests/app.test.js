import React from 'react';
import App from '../App'
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'

const middlewares = []
const mockStore = configureStore(middlewares)
const store = mockStore({routing: {locationBeforeTransitions: {pathname: '/signup'}}});

test('App test', () => {
  const component = renderer.create(<Provider store={store}><App /></Provider>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
