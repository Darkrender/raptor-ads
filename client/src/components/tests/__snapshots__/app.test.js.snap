exports[`test App test 1`] = `
<div>
  <div
    className="ui inverted segment">
    <div
      className="ui inverted pointing secondary menu">
      <div
        className="item logo"
        onClick={[Function]}>
        <img
          src="/client/src/assets/half-raptor.png" />
      </div>
      <a
        className="item"
        onClick={[Function]}
        style={Object {}}>
        About
      </a>
      <a
        className="item"
        onClick={[Function]}
        style={Object {}}>
        Customer Dashboard
      </a>
      <div
        className="right menu">
        <a
          className="item"
          onClick={[Function]}
          style={Object {}}>
          Sign Up
        </a>
        <a
          className="item"
          onClick={[Function]}
          style={Object {}}>
          Login
        </a>
      </div>
    </div>
  </div>
</div>
`;
